scvi/model/_totalvi.py

```diff
@@ -1045,6 +1046,79 @@ def _validate_anndata(
 
         return adata
 
+    def _get_totalvi_protein_priors(self, adata, n_cells=100):
+        """Compute an empirical prior for protein background."""
+        import warnings
+
+        from sklearn.exceptions import ConvergenceWarning
+        from sklearn.mixture import GaussianMixture
+
+        warnings.filterwarnings("error")
+
+        adata = self._validate_anndata(adata)
+        batch = self.get_from_registry(adata, _CONSTANTS.BATCH_KEY).ravel()
+        cats = adata.uns["_scvi"]["categorical_mappings"]["_scvi_batch"]["mapping"]
```

Comment:
is this safe now?

Reply:
no I haven't moved all these references yet, I'm planning on doing that in chunks